pkg/fuzzy/google_translate.go

```diff
@@ -160,27 +142,74 @@ func (g *GoogleTranslate) Translate(
 			Contents:           batches[i],
 		})
 		if err != nil {
-			return nil, fmt.Errorf("google translate client: translate texts from batch #%d: %w", i, err)
+			return nil, fmt.Errorf("google translate client: translate text #%d from batch: %w", i, err)
+		}
+
+		for i := range res.GetTranslations() {
+			translatedText = append(translatedText, res.GetTranslations()[i].GetTranslatedText())
 		}
+	}
+
+	// Overwrite text nodes in ASTs to include newly translated text.
+	if err := textNodes.OverwriteText(translatedText); err != nil {
+		return nil, fmt.Errorf("google translate: overwrite text nodes in ASTs: %w", err)
+	}
 
-		for _, t := range res.GetTranslations() {
-			m := translation.Messages[msgIndex]
-			m.Message = t.GetTranslatedText()
-			m.Status = model.MessageStatusFuzzy
+	// create translation with newly translated messages.
+	translated := model.Translation{
+		Language: targetLanguage,
+		Original: translation.Original,
+		Messages: make([]model.Message, len(translation.Messages)),
+	}
 
-			translated.Messages = append(translated.Messages, m)
+	for i := range asts {
+		var err error
 
-			msgIndex++
+		m := translation.Messages[i]
```

Comment:
Are you certain the index is not out of range?

Reply:
since ast is created for each ranslation.Message, number of asts should match with number of translation.Messages.